language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/TypeGuardCodeAction.java

```diff
@@ -175,10 +173,34 @@ public String getName() {
         SemanticModel semanticModel = context.currentSemanticModel().orElseThrow();
         Document srcFile = context.currentDocument().orElseThrow();
         Optional<Symbol> symbol = semanticModel.symbol(srcFile,
-                                                       assignmentStmtNode.varRef().lineRange().startLine());
+                assignmentStmtNode.varRef().lineRange().startLine());
         if (symbol.isEmpty() || symbol.get().kind() != SymbolKind.VARIABLE) {
             return Optional.empty();
         }
         return Optional.of((VariableSymbol) symbol.get());
     }
+
+    private boolean isInValidContext(NonTerminalNode matchedNode, CodeActionContext context) {
+        Optional<SyntaxTree> syntaxTree = context.currentSyntaxTree();
+        if (syntaxTree.isEmpty()) {
+            return false;
+        }
+        if (matchedNode.kind() == SyntaxKind.ASSIGNMENT_STATEMENT) {
+            Node varRef = ((AssignmentStatementNode) matchedNode).varRef();
+            return isCursorWithinIdentifier(varRef, context);
+        }
+        if (matchedNode.kind() == SyntaxKind.LOCAL_VAR_DECL) {
+            BindingPatternNode bindingPatternNode = 
+                    ((VariableDeclarationNode) matchedNode).typedBindingPattern().bindingPattern();
+            return isCursorWithinIdentifier(bindingPatternNode, context);
+        }
+        return false;
+    }
+
+    private boolean isCursorWithinIdentifier(Node node, CodeActionContext context) {
+        LinePosition cursorPosition = 
+                LinePosition.from(context.cursorPosition().getLine(), context.cursorPosition().getCharacter());
+        int cursorPosOffset = context.currentSyntaxTree().get().textDocument().textPositionFrom(cursorPosition);
```

Comment:
in this scenario we have already checked for `context.currentSyntaxTree().isPresent()` in line 185. But if we're using the function again, we will have to do the check again.